src/eel2_adapter.h

```diff
@@ -42,6 +53,63 @@ class EEL2Adapter {
   double **mInputs = nullptr;
   double **mOutputs = nullptr;
 
+  World *mWorld;
+  Graph *mParent;
+
+  // cache the latest sndbuf b/c it is likely that we
+  // stick to one sndbuf
+  SndBuf *mSndBuf = nullptr;
+  int mSndBufNum = -1;
+
   // @todo make this a RT alloc char* or free this via NRT thread
   std::string mScript;
+
+  // see GET_BUF macro from SC_Unit.h
+  std::optional<SndBuf*> GetBuffer(int bufNum) {
+    if (bufNum < 0) {
+      return {};
+    };
+    if (bufNum == mSndBufNum && mSndBuf != nullptr) {
+      return mSndBuf;
+    }
+
+    // de-validate cache
+    mSndBuf = nullptr;
+    mSndBufNum = bufNum;
+
+    if (bufNum < mWorld->mNumSndBufs) {
+      mSndBuf = mWorld->mSndBufs + bufNum;
+      return mSndBuf;
+    };
+
+    // looking for a matching localbuf
+    int localBufNum = bufNum - mWorld->mNumSndBufs;
+    if (localBufNum <= mParent->localBufNum) {
+      mSndBuf = mParent->mLocalSndBufs + localBufNum;
+      return mSndBuf;
+    }
+
+    // no buffer found
+    mSndBuf = nullptr;
+    return {};
+  }
+
+  // Assumes that chan is within bounds
+  static float GetSample(const SndBuf *buf, const int chan, const int sampleNum) {
+    LOCK_SNDBUF_SHARED(buf);
```

Comment:
1. why PascalCase? 

2. don't use `const` for parameters that are passed by value. The signature should be:
```
float getSample(const SndBuf *buf, int chan, int sampleNum) {
```

3. I would just clamp `sampleNum` and then return the sample:
```
sampleNum = std::clamp<int>(sampleNum, 0, buf->samples - 1);
return buf->data[buf->channels * sampleNum + chan];
```
Shorter and easier to understand.